src/classes/User.js

```diff
@@ -55,6 +55,37 @@ class User {
     const filteredRecipe = this.repo.findRecipesByIngredient(ingredientName, this.favoriteRecipes);
     return filteredRecipe;
   }
+
+  returnPantryIngredients() {
+    let matchingIngredient = 'banana';
+
+    return this.pantry.reduce((acc, pantryIngredient) => {
+      this.repo.recipes.find(recipe => {
+        matchingIngredient =  recipe.ingredients.find(ingredient => pantryIngredient.ingredient === ingredient.id)
+
+        return matchingIngredient;
+      })
+
+      acc.push(`${pantryIngredient.amount} ${matchingIngredient.name}`)
+      return acc
+    }, [])
+  }
+
+  updateIngredientAmount(ingredients) {
+    ingredients.forEach(ingredient => {
+      let match = this.pantry.find(pantryIngredient => pantryIngredient.ingredient === ingredient.id)
+
+      match.amount += ingredient.amount;
+      if (!match.amount) {
+        this.removeIngredient(match);
+      }
```

Comment:
So, the logic here, if I follow is this:

- The matching pantry ingredient's amount is updated based on the amount needed for the recipe
- If the matching pantry ingredient's amount is falsy, we remove it (would have to be `0` because `!-1` is truth)

If this sounds right, I have a few questions/comments:

- What happens if `match.amount` is negative? Is this even possible? How do you ensure it's not possible?
- It seems like `updateIngredientAmount` is being used for both adding to and removing from the pantry. As a developer without any knowledge of how this is being used, it's not immediately clear how this logic works for both scenarios. I would consider a refactor to have single responsibility functions for add/remove at some point and make helper functions to limit any redundant code.

Reply:
I think our first check of the pantry should avoid having negative numbers, but I am not 100% sure. Definitely something to pay attention to. I will think of alternative ways to check that.

Reply:
I separated the add and subtract functions. At the beginning I thought that maybe we would want to both add to an ingredient and subtract from another, but that should never be the case since we only add or subtract from them at one point.